lib/mobility/plugins/active_model/dirty.rb

```diff
@@ -75,23 +124,157 @@ def included(model_class)
           # they change from Rails version to version.
           def method_suffixes
             @method_suffixes ||=
-              Class.new do
-                include ::ActiveModel::Dirty
-              end.attribute_method_matchers.map(&:suffix).select { |m| m =~ /\A_/ }
+              begin
+                klass = Class.new { include ::ActiveModel::Dirty }
+                klass.attribute_method_matchers.map(&:suffix).select do |m|
+                  # only include suffixes for non-private handler methods
+                  m =~ /\A_/ && !klass.private_instance_methods.include?(:"attribute#{m}")
+                end
```

Comment:
@pwim So in order to support all these attribute methods, I need to override them explicitly... which is really complicated because each version of Rails supports different methods. What I've done (which I was doing previously) is to figure out the methods by including `::ActiveModel::Dirty` into a class and then checking its suffixes... in this case though, I only want to define the ones that are *public*, which requires an extra check.

The result, though, is that at least we don't have a ton of conditionals in the code, just in the specs.

I could alternatively define the `attribute_` methods and have the suffix methods call them, rather than the other way around... but in the end I don't think it really matters. Main thing is that tests cover all cases well.

Reply:
Actually not done yet... have to do the same for suffixes defined in Active**Record**...

Reply:
Oh rats, this doesn't even work actually. This would support `article.attribute_changed?(:title)`, but previous behaviour was `article.attribute_changed?(:title_en)`... need to rethink.